pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -257,14 +255,13 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 		updateCleanupPolicy = kubectl.RenameRollingUpdateCleanupPolicy
 	}
 	config := &kubectl.RollingUpdaterConfig{
-		Out:            out,
-		OldRc:          oldRc,
-		NewRc:          newRc,
-		UpdatePeriod:   period,
-		Interval:       interval,
-		Timeout:        timeout,
-		CleanupPolicy:  updateCleanupPolicy,
-		UpdateAcceptor: kubectl.DefaultUpdateAcceptor,
+		Out:           out,
+		OldRc:         oldRc,
+		NewRc:         newRc,
+		UpdatePeriod:  period,
+		Interval:      interval,
+		Timeout:       timeout,
+		CleanupPolicy: updateCleanupPolicy,
```

Comment:
I doubt anyone would want a rolling update to take down all their pods
(because they take longer to start than it does for the entire rolling
update to schedule the pods).  The abstraction you're looking for is how to
determine when something meets the availability minimum - the default might
be "readiness", but there is also "running" or "created"

On Tue, Jul 28, 2015 at 5:01 PM, Dan Mace <notifications@github.com> wrote:

> In pkg/kubectl/cmd/rollingupdate.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11942#discussion_r35700339>
> :
>
> > @@ -257,14 +255,13 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
> >  		updateCleanupPolicy = kubectl.RenameRollingUpdateCleanupPolicy
> >  	}
> >  	config := &kubectl.RollingUpdaterConfig{
> > -		Out:            out,
> > -		OldRc:          oldRc,
> > -		NewRc:          newRc,
> > -		UpdatePeriod:   period,
> > -		Interval:       interval,
> > -		Timeout:        timeout,
> > -		CleanupPolicy:  updateCleanupPolicy,
> > -		UpdateAcceptor: kubectl.DefaultUpdateAcceptor,
> > +		Out:           out,
> > +		OldRc:         oldRc,
> > +		NewRc:         newRc,
> > +		UpdatePeriod:  period,
> > +		Interval:      interval,
> > +		Timeout:       timeout,
> > +		CleanupPolicy: updateCleanupPolicy,
>
> Hm. The impact of this PR on the default would be that newly scaled pods
> will need to become ready to make progress, whereas today progress will be
> made as long as pods are *scheduled* (regardless of readiness).
>
> Assuming this new behavior is even acceptable in general, we'll need to
> decide whether to preserve today's looser scaling semantics, perhaps by
> only enforcing availability when updatePercent is specified (for example).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11942/files#r35700339>
> .
>



-- 
Clayton Coleman | Lead Engineer, OpenShift